test/e2e/metrics/editor.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { readFileSync } from 'fs';
+import path from 'path';
+import {
+	DataHelper,
+	envVariables,
+	EditorPage,
+	TestAccount,
+	getTestAccountByFeature,
+	envToFeatureKey,
+	PostsPage,
+	Metrics,
+} from '@automattic/calypso-e2e';
+import { Browser, Page } from 'playwright';
+
+declare const browser: Browser;
+const results: Record< string, number[] > = {};
+
+describe( DataHelper.createSuiteTitle( 'Metrics: Editor' ), function () {
+	const features = envToFeatureKey( envVariables );
+	const accountName = getTestAccountByFeature(
+		features,
+		// The default accounts for gutenberg+simple are `gutenbergSimpleSiteEdgeUser` for GB edge
+		// and `gutenbergSimpleSiteUser` for stable. The criteria below conflicts with the default
+		// one that would return the `gutenbergSimpleSiteUser`. We also can't define it as part of
+		// the default criteria, and should pass it here, as an override. For this specific function
+		// call, `simpleSitePersonalPlanUser` will be retured when gutenberg is stable, and siteType
+		// is simple.
+		[ { gutenberg: 'stable', siteType: 'simple', accountName: 'simpleSitePersonalPlanUser' } ]
+	);
+
+	let page: Page;
+	let editorPage: EditorPage;
+	let postsPage: PostsPage;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+
+		const testAccount = new TestAccount( accountName );
+		await testAccount.authenticate( page );
+	} );
+
+	it( 'Start and fill a test post', async function () {
+		postsPage = new PostsPage( page );
+		const metrics = new Metrics( page );
+		await postsPage.visit();
+		await postsPage.newPost();
+		editorPage = new EditorPage( page );
+		await editorPage.waitUntilLoaded();
+		const filePath = path.join( __dirname, './fixtures/large-post.html' );
+		await editorPage.loadHtmlContent( readFileSync( filePath, 'utf8' ).trim() );
+		const canvas = await editorPage.getEditorCanvas();
+		await canvas.locator( '.wp-block' ).first().waitFor();
+		await editorPage.enterTitle( 'Test Post' );
+		await editorPage.publish();
+
+		const samples = 2;
+		const throwaway = 1;
+		const iterations = samples + throwaway;
+
+		for ( let i = 1; i <= iterations; i++ ) {
+			await page.reload();
+			editorPage = new EditorPage( page );
+			editorPage.waitUntilLoaded();
+			const canvas = await editorPage.getEditorCanvas();
+			// Wait for the first block.
+			await canvas.locator( '.wp-block' ).first().waitFor();
+			// Get the durations.
+			const loadingDurations = await metrics.getLoadingDurations();
+
+			// Save the results.
+			if ( i > throwaway ) {
+				Object.entries( loadingDurations ).forEach( ( [ metric, duration ] ) => {
+					const metricKey = metric === 'timeSinceResponseEnd' ? 'firstBlock' : metric;
+					if ( ! results[ metricKey ] ) {
+						results[ metricKey ] = [];
+					}
+					results[ metricKey ].push( duration );
+				} );
+			}
+		}
+
+		console.log( results );
```

Comment:
I have a question in the PR description about this. I would love to use the "performance-reporter" from either Gutenberg or Woo (check the Woo PR) instead of console logging. It's just a way to dump the results for now.

I've spent some time trying to wire the performance reporter but failed, It needs to be added to the playwright config file but in calypso e2e tests, there's not playwright config file really, it's kind of buried in a series of scripts and configs.

Reply:
Yep, the entire setup is a bit rudimentary, and while it was migrated to Playwright, it still bears the mark of old technology used before it 😅 

This is the config file: https://github.com/Automattic/wp-calypso/blob/2b4ed1bcc9a77c61f4456a70b051200957a84fd7/packages/calypso-e2e/src/jest-playwright-config/index.js - as you can see it uses [Jest Playwright](https://github.com/playwright-community/jest-playwright) as a test runner and was never migrated to `@playwright/test`. 

Now, I haven't tried using a custom reporter there, but you should be able to hook it the same way. You might need to work with a separate config to be able to inherit the original config and specify the custom reporter. Try it out and let me know how it goes. 

Alternatively, migrating to `@playwright/test` could make sense, although I'm not sure what challenges you might bump into and how long it would take. I bet it's outside of your initial scope 😉